src/StoreApi/Routes/AbstractCartRoute.php

```diff
@@ -56,6 +56,14 @@ public function get_response( \WP_REST_Request $request ) {
 		$this->cart_controller->load_cart();
 		$this->calculate_totals();
 
+		if ( $this->requires_nonce( $request ) ) {
+			$nonce_check = $this->check_nonce( $request );
+
+			if ( is_wp_error( $nonce_check ) ) {
+				return $this->add_nonce_headers( $this->error_to_response( $nonce_check ) );
+			}
+		}
```

Comment:
I'm curious about why you moved this out of the `validate_callback` function and eliminated that function?

Reply:
I found that when you use these validation callbacks, WP does the error handling for the response which prevents us from appending updated nonce headers (unless you use a filter hook). I moved it back here so we can control the response.